### ISSUE TITLE
netlify redirects

### ISSUE THREAD

**Issue by boldstar** [gridsome/gridsome#361]
I am trying to use a _redirects file for netlify. My question is what should the redirect rules be? In a SPA I only had to do the following rule
```
/* /index.html 200
```

However this does not work for my gridsome site. Is there a recommended way that redirect rules should be written?

To add more context, this may not necessarily be an issue with my _redirects file. I believe something is throwing an error when trying to navigate to another page. I am referencing this [#289](https://github.com/gridsome/gridsome/issues/289)

So as suggested, I add the following code to my gridsome.config.js file
```
chainWebpack (config) {
    config.mode('development')
  }
```
and this is the alarm I get in the console
```
vue.runtime.esm.js:5190 Uncaught (in promise) DOMException: Failed to execute 'appendChild' on 'Node': This node type does not support this method.
    at Object.appendChild (webpack:///./node_modules/vue/dist/vue.runtime.esm.js?:5190:8)
    at insert (webpack:///./node_modules/vue/dist/vue.runtime.esm.js?:5501:17)
    at createElm (webpack:///./node_modules/vue/dist/vue.runtime.esm.js?:5428:7)
    at createChildren (webpack:///./node_modules/vue/dist/vue.runtime.esm.js?:5510:9)
    at hydrate (webpack:///./node_modules/vue/dist/vue.runtime.esm.js?:5838:11)
    at hydrate (webpack:///./node_modules/vue/dist/vue.runtime.esm.js?:5853:34)
    at hydrate (webpack:///./node_modules/vue/dist/vue.runtime.esm.js?:5853:34)
    at VueComponent.patch [as __patch__] (webpack:///./node_modules/vue/dist/vue.runtime.esm.js?:5929:17)
    at VueComponent.Vue._update (webpack:///./node_modules/vue/dist/vue.runtime.esm.js?:3588:19)
    at VueComponent.updateComponent (webpack:///./node_modules/vue/dist/vue.runtime.esm.js?:3676:10)
appendChild @ vue.runtime.esm.js:5190
insert @ vue.runtime.esm.js:5501
createElm @ vue.runtime.esm.js:5428
createChildren @ vue.runtime.esm.js:5510
hydrate @ vue.runtime.esm.js:5838
hydrate @ vue.runtime.esm.js:5853
hydrate @ vue.runtime.esm.js:5853
patch @ vue.runtime.esm.js:5929
Vue._update @ vue.runtime.esm.js:3588
updateComponent @ vue.runtime.esm.js:3676
get @ vue.runtime.esm.js:4062
Watcher @ vue.runtime.esm.js:4051
mountComponent @ vue.runtime.esm.js:3683
Vue.$mount @ vue.runtime.esm.js:7819
init @ vue.runtime.esm.js:2816
hydrate @ vue.runtime.esm.js:5827
hydrate @ vue.runtime.esm.js:5853
hydrate @ vue.runtime.esm.js:5853
patch @ vue.runtime.esm.js:5929
Vue._update @ vue.runtime.esm.js:3588
updateComponent @ vue.runtime.esm.js:3676
get @ vue.runtime.esm.js:4062
Watcher @ vue.runtime.esm.js:4051
mountComponent @ vue.runtime.esm.js:3683
Vue.$mount @ vue.runtime.esm.js:7819
init @ vue.runtime.esm.js:2816
hydrate @ vue.runtime.esm.js:5827
hydrate @ vue.runtime.esm.js:5853
hydrate @ vue.runtime.esm.js:5853
patch @ vue.runtime.esm.js:5929
Vue._update @ vue.runtime.esm.js:3588
updateComponent @ vue.runtime.esm.js:3676
get @ vue.runtime.esm.js:4062
Watcher @ vue.runtime.esm.js:4051
mountComponent @ vue.runtime.esm.js:3683
Vue.$mount @ vue.runtime.esm.js:7819
init @ vue.runtime.esm.js:2816
hydrate @ vue.runtime.esm.js:5827
patch @ vue.runtime.esm.js:5929
Vue._update @ vue.runtime.esm.js:3588
updateComponent @ vue.runtime.esm.js:3676
get @ vue.runtime.esm.js:4062
Watcher @ vue.runtime.esm.js:4051
mountComponent @ vue.runtime.esm.js:3683
Vue.$mount @ vue.runtime.esm.js:7819
init @ vue.runtime.esm.js:2816
hydrate @ vue.runtime.esm.js:5827
patch @ vue.runtime.esm.js:5929
Vue._update @ vue.runtime.esm.js:3588
updateComponent @ vue.runtime.esm.js:3676
get @ vue.runtime.esm.js:4062
Watcher @ vue.runtime.esm.js:4051
mountComponent @ vue.runtime.esm.js:3683
Vue.$mount @ vue.runtime.esm.js:7819
(anonymous) @ entry.client.js:54
(anonymous) @ vue-router.esm.js:1819
(anonymous) @ vue-router.esm.js:1819
(anonymous) @ vue-router.esm.js:1921
step @ vue-router.esm.js:1651
step @ vue-router.esm.js:1658
runQueue @ vue-router.esm.js:1662
(anonymous) @ vue-router.esm.js:1916
step @ vue-router.esm.js:1651
(anonymous) @ vue-router.esm.js:1655
(anonymous) @ vue-router.esm.js:1901
(anonymous) @ vue-router.esm.js:1694
(anonymous) @ vue-router.esm.js:1770
Promise.then (async)
(anonymous) @ vue-router.esm.js:1717
(anonymous) @ vue-router.esm.js:1738
(anonymous) @ vue-router.esm.js:1738
flatMapComponents @ vue-router.esm.js:1737
(anonymous) @ vue-router.esm.js:1673
iterator @ vue-router.esm.js:1880
step @ vue-router.esm.js:1654
step @ vue-router.esm.js:1658
runQueue @ vue-router.esm.js:1662
confirmTransition @ vue-router.esm.js:1909
transitionTo @ vue-router.esm.js:1811
init @ vue-router.esm.js:2428
beforeCreate @ vue-router.esm.js:532
invokeWithErrorHandling @ vue.runtime.esm.js:1702
callHook @ vue.runtime.esm.js:3825
Vue._init @ vue.runtime.esm.js:4496
Vue @ vue.runtime.esm.js:4570
createApp @ app.js:90
(anonymous) @ entry.client.js:22
./node_modules/gridsome/app/entry.client.js @ app.12303bfc.js:2530
__webpack_require__ @ app.12303bfc.js:84
(anonymous) @ entry.client.js:1
0 @ app.12303bfc.js:3431
__webpack_require__ @ app.12303bfc.js:84
checkDeferredModules @ app.12303bfc.js:46
(anonymous) @ app.12303bfc.js:218
(anonymous) @ app.12303bfc.js:221
Show 37 more frames
vue.runtime.esm.js:1734 TypeError: Cannot read property '_isDestroyed' of undefined
    at destroy (vue.runtime.esm.js:2855)
    at invokeDestroyHook (vue.runtime.esm.js:5571)
    at invokeDestroyHook (vue.runtime.esm.js:5576)
    at invokeDestroyHook (vue.runtime.esm.js:5576)
    at invokeDestroyHook (vue.runtime.esm.js:5576)
    at invokeDestroyHook (vue.runtime.esm.js:5576)
    at invokeDestroyHook (vue.runtime.esm.js:5576)
    at VueComponent.patch [as __patch__] (vue.runtime.esm.js:5903)
    at VueComponent.Vue.$destroy (vue.runtime.esm.js:3644)
    at destroy (vue.runtime.esm.js:2857)
```

**Comment by danvega** [comment]
What type of redirect rule are you trying to create? I was able to create a few redirects with no issue.

```
# 301 Redirects
/blog/creating-your-first-npm-package    /blog/2019/02/10/creating-your-first-npm-package   301
/blog/hello-gridsome /blog/2019/01/31/hello-gridsome 301

# pretty urls
/feed /rss.xml 301
```

**Comment by robaxelsen** [comment]
@boldstar :point_up:

**Comment by boldstar** [comment]
@danvega & @robaxelsen , sorry for the late response. Actually my issue was not redirects at all. I was using netlify forms and it was throwing an error I was not aware of. Once I fixed that I had no issues. Thank you for the help!

**Comment by JosephBlythe** [comment]
@boldstar Hey I have a similar issue with that error also using netlify forms, could you explain what was causing the error?

**Comment by hacknug** [comment]
@JosephBlythe I remember having issues with Netlify Forms too and I think this fixed it: https://github.com/hacknug/klayaya/commit/9b0442e9394ecc4d8fe205c4a82a2e1dc3156014

**Comment by snoopdouglas** [comment]
For those looking to specify redirects from eg. Netlify CMS and create a `_redirects` as part of their deployment, we've come up with a (very barebones) solution.

Netlify CMS writes a redirects.yml for a single-file "Redirects" collection, a la:

```js
redirects:
  - from: /a
     to: /b
     temporary: false
  - from: /c
     to: /d
     temporary: true
```

The following script then runs pre-deployment, after which the resultant `_redirects` file is moved into `dist`.

```js
#!/usr/bin/env node

const yaml = require('js-yaml');
const fs = require('fs');

const file = fs.readFileSync('redirects.yml');
const parsed = yaml.safeLoad(file);

const code = temp => temp ? '302' : '301';
let out = parsed.redirects.map((r) => `${r.from} ${r.to} ${code(r.temporary)}`).join('\n');

fs.writeFileSync('_redirects', out);
```

**Comment by JosephBlythe** [comment]
> @JosephBlythe I remember having issues with Netlify Forms too and I think this was what fixed it: [hacknug/klayaya@9b0442e](https://github.com/hacknug/klayaya/commit/9b0442e9394ecc4d8fe205c4a82a2e1dc3156014)

@hacknug Thank you adding the hidden input with the form-name did the trick.

**Comment by manavm1990** [comment]
The hidden `<input>` trick works, so thanks for that. [Here](https://docs.netlify.com/forms/setup/#work-with-javascript-rendered-forms) the docs mention having to do this with a JS rendered page, as would be the case for a typical client-side SPA.

But my understanding is that Gridsome is working as a JAMstack - generating static files that we should be able to just upload via traditional FTP, etc.

So, if that's the case, then all of the `<form>` should already be 'rendered' via HTML and be ready for Netlify.

That's Y I didn't think we needed @hacknug solution. 😖
